File: crypto-casino-backend/src/main/java/hu/bme/aut/crypto_casino_backend/model/Transaction.java
package hu.bme.aut.crypto_casino_backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "wallet_id", nullable = false)
    private Wallet wallet;

    @Column(nullable = false)
    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TransactionType type;

    @Column(name = "transaction_hash")
    private String transactionHash;

    @Column(name = "block_number")
    private Long blockNumber;

    @Column(name = "ethereum_amount")
    private BigDecimal ethereumAmount;

    @Column(name = "casino_token_amount")
    private BigDecimal casinoTokenAmount;

    @Column(name = "transaction_time", nullable = false)
    private LocalDateTime transactionTime;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TransactionStatus status;

    public enum TransactionType {
        DEPOSIT,
        WITHDRAWAL,
        EXCHANGE_ETH_TO_TOKEN,
        EXCHANGE_TOKEN_TO_ETH,
        BET,
        WIN
    }

    public enum TransactionStatus {
        PENDING,
        COMPLETED,
        FAILED,
        CANCELLED
    }
}

File: crypto_casino_android/app/src/main/java/hu/bme/aut/crypto_casino_android/data/model/transaction/Transaction.kt
package hu.bme.aut.crypto_casino_android.data.model.transaction

import java.time.ZonedDateTime

data class Transaction(
    val id: Long? = null,
    val userId: Long? = null,
    val walletId: Long? = null,
    val amount: Double = 0.0,
    val type: TransactionType? = null,
    val transactionHash: String? = null,
    val blockNumber: Long? = null,
    val ethereumAmount: Double? = null,
    val casinoTokenAmount: Double? = null,
    val transactionTime: ZonedDateTime? = null,
    val status: TransactionStatus? = null
)

enum class TransactionType {
    DEPOSIT, WITHDRAWAL, PURCHASE, EXCHANGE
}

enum class TransactionStatus {
    PENDING, COMPLETED, FAILED
}
